Sql_Assignment _4_B.txt

```diff
@@ -0,0 +1,80 @@
+-- E-commerce Platform Schema
+
+CREATE TABLE Customers (
+    CustomerID INT PRIMARY KEY,
+    CustomerName VARCHAR(100),
+    Email VARCHAR(100),
+    RegistrationDate DATE
+);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(200),
+    Price DECIMAL(10, 2),
+    CategoryID INT
+);
+
+CREATE TABLE Categories (
+    CategoryID INT PRIMARY KEY,
+    CategoryName VARCHAR(100)
+);
+
+CREATE TABLE Orders (
+    OrderID INT PRIMARY KEY,
+    CustomerID INT,
+    OrderDate DATE,
+    TotalAmount DECIMAL(10, 2),
+    FOREIGN KEY (CustomerID) REFERENCES Customers(CustomerID)
+);
+
+CREATE TABLE OrderDetails (
+    OrderDetailID INT PRIMARY KEY,
+    OrderID INT,
+    ProductID INT,
+    Quantity INT,
+    FOREIGN KEY (OrderID) REFERENCES Orders(OrderID),
+    FOREIGN KEY (ProductID) REFERENCES Products(ProductID)
+);
+
+-- Questions
+
+-- 1. List all products with their category names, including products without a category.
+
+select productname,categoryname from products p left join categories c on p.categoryid=c.categoryid
+
+-- 2. Display all customers and their order history, including customers who haven't placed any orders.
+
+select customername,productname,orderdate,quantity,totalamount from customers c left join orders o on  c.CustomerID=o.CustomerID left join OrderDetails od on od.orderid=o.orderid left join products p on p.ProductID=od.ProductID
+
+-- 3. Show all categories and the products in each category, including categories without any products.
+
+select Categoryname,productname from Categories c left join  Products p on c.categoryid=p.categoryid
+
+-- 4. List all possible customer-product combinations, regardless of whether a purchase has occurred.
+
+select customername,productname from Customers cross join products
+
+-- 5. Display all orders with customer and product information, including orders where either the customer or product information is missing.
+
+select o.orderid,ProductName,c.CustomerID,Quantity,OrderDate,TotalAmount from orders o full join orderdetails od on o.orderid=od.orderid left join products p on od.ProductID=p.ProductID left join customers c on o.CustomerID=c.CustomerID
```

Comment:
FULL JOIN is unnecessary here and could produce unexpected results. LEFT JOINs are sufficient.